formatter/ntopng_json.go

```diff
@@ -0,0 +1,169 @@
+package formatter
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+	"net"
+	"strconv"
+
+	"github.com/netsampler/goflow2/v2/decoders/netflow"
+	protoproducer "github.com/netsampler/goflow2/v2/producer/proto"
+)
+
+type NtopngJson struct {
+}
+
+func (d *NtopngJson) Prepare() error {
+	return nil
+}
+
+func (d *NtopngJson) Init() error {
+	return nil
+}
+
+func (d *NtopngJson) Format(data interface{}) ([]byte, []byte, error) {
+	// The Transport might use "key", but we don't care about it here.
+	var key []byte
+	if dataIf, ok := data.(interface{ Key() []byte }); ok {
+		key = dataIf.Key()
+	}
+
+	ppMsg, ok := data.(*protoproducer.ProtoProducerMessage)
+	if !ok {
+		return key, nil, errors.New("skipping non-ProtoProducerMessage")
+	}
+
+	jdata, err := toJSON(ppMsg)
+	if err != nil {
+		return key, nil, err
+	}
+	return key, jdata, nil
+}
+
+/*
+ * Converts a FlowMessage to JSON for ntopng
+ *
+ * TODO: Figure out how to get remapped IN/OUT bytes/pkts here. This still uses original
```

Comment:
this todo seems out of date??

Reply:
For this PR, it's not. I have a diff PR that actually makes the IN/OUT_BYTES available. As of this PR, Bytes sent to ntopng are still 0. 

Do you want me to add that PR to this one? I figured this was big enough as is.